src/Page/PageManager.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace V17Development\FlarumSeo\Page;
+
+use Flarum\Extension\ExtensionManager;
+use V17Development\FlarumSeo\Page\PageDriverInterface;
+use V17Development\FlarumSeo\SeoExtenderManagerInterface;
+
+class PageManager implements SeoExtenderManagerInterface
+{
+    /**
+     * @var array
+     */
+    protected $drivers = [];
+
+    /**
+     * @var ExtensionManager
+     */
+    protected $extensionManager;
+
+    /**
+     * @param ExtensionManager $extensionManager
+     */
+    public function __construct(ExtensionManager $extensionManager)
+    {
+        $this->extensionManager = $extensionManager;
+    }
+
+    public function addDriver(string $name, PageDriverInterface $driver)
+    {
+        $this->drivers[$name] = $driver;
+    }
+
+    public function getDrivers(string $routeName = null)
+    {
+        // Filter available drivers
+        return array_filter($this->drivers, function (PageDriverInterface $driver) use ($routeName) {
+            // Filter drivers that require extensions to be enabled
+            foreach ($driver->extensionDependencies() as $extensionId) {
+                if (!$this->extensionManager->isEnabled($extensionId)) {
+                    return false;
+                }
+            }
+
+            // Check if driver should be handled
+            if (!is_null($routeName) && count($driver->handleRoutes()) !== 0 && !in_array($routeName, $driver->handleRoutes())) {
+                return false;
+            }
+
+            return true;
+        });
```

Comment:
This cannot be inside this one method, you can see the code smell from the level of indentation as well as the need for comments. Clean, understandable code is best written by keeping methods as small and focused on one responsibility.

Split this code up so that you have one method: `getActiveDrivers` which returns a `Collection` filtered by drivers of which the extension dependencies are enabled.

Then this method can `filter` based on the route:

```suggestion
        return $this->getActiveDrivers()
            ->filter(fn (PageDriverInterface $driver) => $routeName === null || in_array($routeName, $driver->handleRoutes() ?? []))
            ->toArray();
```

Untested ofcourse ;)

Reply:
Arrow functions aren't supported in PHP 7.3, I don't think?

Reply:
Ah, I thought the minimum PHP version was 8 and higher. I guess I've mistaken it with Symfony 's minimum requirements instead 👀 

Will fix it 👍

Reply:
@davwheat 7.3 isn't even one of the supported versions anymore. This means any critical vulnerability will not be patched 🎉 

https://www.php.net/supported-versions.php

Reply:
That's fair. I just assumed we'd still want to support it as Flarum still does.

Reply:
I think core is stuck to 7.3 because we can't just increase it between minor versions. I think 2.0 and L9/10 will bring a minimum of 8+ anyway.